subsys/zigbee/lib/zigbee_fota/src/zigbee_fota.c

```diff
@@ -202,7 +202,7 @@ static uint8_t ota_dfu_target_init(const uint8_t *magic_word_buf)
 		return err;
 	}
 
-	err = dfu_target_init(img_type, ota_ctx.bin_size,
+	err = dfu_target_init(img_type, 0, ota_ctx.bin_size,
```

Comment:
Is the image number stored somewhere inside the upgrade binaries?
If so, it may be quite easy to avoid this hard-coded value.

Reply:
Currently I don't think this metadata exsists. For multi image on `nrf53` one could maybe try to look at the image header's reset address to select which slot it's supposed to go into by using logic something like `address > PM_NETCORE_START_ADDRESS` This would not work universaly.

Reply:
Are there any plans to add a unified metadata with this kind of information, that will be interpreted by the `dfu_target`, not by the protocol implementation?

I prefer to look on the protocol implementation as on the binary pipe, that does not necessary know what kind of update it downloads.
The Zigbee spec itself allows to pass several images within the payload, so the current approach should be extendable to cover multiple image numbers, but this is outside of the scope of this PR.